src/client/components/auth-hoc/index.jsx

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { connect } from "react-redux";
+import { mapStateToProps, mapDispatchToProps } from "./container";
+
+class AuthHOC extends React.Component {
+    componentWillMount() {
+        this.props.getCurrentUser();
+    }
+
+    render() {
+        const { component, user } = this.props;
+        if (!user) {
+            return null;
```

Comment:
It's better if it shows "Please log in to see the page" with a Login button.

Reply:
it redirects to login, when userService.getCurrentUser finds nothing